Lib/fontTools/otlLib/builder.py

```diff
@@ -2801,14 +2801,10 @@ def _addName(nameTable, value, minNameID=0):
     elif isinstance(value, dict):
         names = value
     elif isinstance(value, list):
-        nameID = nameTable._findUnusedNameID()
-        for nameRecord in value:
-            if all(hasattr(nameRecord, attr) for attr in
-                   ["string", "nameID", "platformID", "platEncID", "langID"]):
-                nameTable.setName(nameRecord.string,
-                                  nameID,nameRecord.platformID,
-                                  nameRecord.platEncID,nameRecord.langID)
-        return nameID
+        names = {}
+        for name in value:
+            langCode = _WINDOWS_LANGUAGES.get(name.langID)
```

Comment:
hm.. here you're assuming that the namerecord is a windows-platform one and you're only using its langID, i don't think you can make this assumption. 
it kind of was more correct before this last edit... You were actually right, we don't want to use `addMultilingualName` when we already have a list of NameRecords.

What we want to do here is basically append the name records, it turns out that the name table has an `addName` method that does exactly what you want: i.e. adds a string with given triplet encoding and assigns it to an unused nameID slot in the user range > 255.
Just use that instead of the combo _findUnusedNameID + setName.

Also, when _addName gets a list, make sure that all the values are instances of `_n_a_m_e.NameRecord` (raise TypeError if they aren't). Document that in the buildStatTable docstring that `elidedFallbackName` can also be a list of namerecords (in addition to a int or a dict).

Reply:
@anthrotype I tried using `addName` where it iterates through `value` and is called for each name. However, I realized it gives a new nameID for each name every time, which creates a problem for an ElidedFallBackName with multiple names. For example:

`table STAT {
    ElidedFallbackName {
        name "Roman";
        name 3 1 1041 "ローマン";
    };`

I was going to suggest this, which still uses `addMultilingualName` but specifies which platform is being requested, and keeps the nameID the same for multiple names in ElidedFallbackName:

`        names = {}
        mac = False
        windows = False
        for name in value:
            if name.platformID == 1:
                langCode = _MAC_LANGUAGES.get(name.langID)
                mac = True
            else:
                langCode = _WINDOWS_LANGUAGES.get(name.langID)
                windows = True
            names[langCode] = name.string
        return nameTable.addMultilingualName(names, windows=windows, mac=mac, minNameID=minNameID)`

Reply:
I see. Ok, then I suggest we revert to using setName like you were doing originally. It's probably fine that you use `_findUnusedNameID` even if underscore-prefixed and intended to be private.